tck/src/main.cc

```diff
@@ -0,0 +1,55 @@
+/*-
+ *
+ * Hedera C++ SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "JsonTypeMapper.h"
+#include "SdkClient.h"
+#include "TckServer.h"
+#include "impl/EntityIdHelper.h"
+
+using namespace Hedera::TCK;
+
+int main(int argc, char** argv)
+{
+  // Initialize the TCK server with the port number.
+  TckServer tckServer((argc > 1) ? static_cast<int>(Hedera::internal::EntityIdHelper::getNum(argv[1])) : // NOLINT
+                        TckServer::DEFAULT_HTTP_PORT);
+
+  // Add the SDK client functions.
+  tckServer.add("createAccount",
```

Comment:
Just a little question here. Will it be better for the `tckServer` to resolve the `getHandles`? I can see that most of the time the `.add` takes similar names for the first argument which is name and then calls `getHandle` for the same name? And just was wondering shouldn't we define the SDK client functions in the `tckServer` not having to add them every time one by one or is the idea here to have like different suites of tests and that is why you decided to leave it so we can add different SDK functions independently?

Reply:
I styled a lot of this implementation off of the `json-rpc-cxx` [library](https://github.com/jsonrpcx/json-rpc-cxx) so didn't think too much deeper about the implementation here. Since C++ doesn't support reflection, there needs to be some sort of mapping from the function name string to the actual function. The `getHandle` function also helps differentiate between JSON RPC methods and notifications, which we may or may not use notifications ever but the functionality doesn't hurt to have I think.

As for `SdkClient`, the idea was to have all SDK-facing functionality in its own object (or namespace in this matter, since `SdkClient` is acting like a singleton). But yes, it would be a good idea at some point when more tests are added to possibly make up different test suites, which would also provide more organization to the code.

Reply:
Nice we can keep the test suite thing as a future idea.